### PR TITLE
Adding URL params management

### PR DESCRIPTION
**Pull request by pakb**
Adding management of params lon/lat/zoom

[Test link](https://web-mapviewer.dev.bgdi.ch/feature_BGDIINF_SB-1162_url_param_mngt/index.html)

### CODE REVIEW

File: src/modules/menu/components/MenuLangSelector.vue [hunk 1]
```diff
@@ -8,7 +8,7 @@
 </style>
 
 <script>
-  import LangSwitchToolbar from "@/modules/i18n/components/LangSwitchToolbar";
+  import LangSwitchToolbar from "../../i18n/components/LangSwitchToolbar";
```
Comment (boecklic) [inline src/modules/menu/components/MenuLangSelector.vue:11]: out of curiosity: to me the `@/modules/...` way of addressing files looks cleaner than the `../../...` way, what's the reason for the change?

**Review by boecklic — CHANGES_REQUESTED** [review]
nice! I like the clean separation of the `*.api.js` part, leaves us flexibility to change those API's in the future.

File: src/modules/menu/components/MenuLangSelector.vue (hunk 1 above)
Comment (pakb) [reply src/modules/menu/components/MenuLangSelector.vue:11]: Some of them where changed because otherwise I couldn't the store in isolation in Cypress. This `@/` notation is handled by Vue and webpack, but if left as is and then imported straight into a `..._spec.js` file from Cypress, Cypress can't handle the `@` part (I read it's possible to configure something somewhere, I also prefer the Vue notation I could give it a look)

File: src/router/legacy-permalink-management.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import proj4 from "proj4";
+import {round} from "../numberUtils";
+
+const legacyZoomMapping = {
```
[the remark this answers is not in the document]
Comment (pakb) [reply src/router/legacy-permalink-management.js:4]: good idea, that would clarify a bit more what it is!

File: src/api/layers.api.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                    reject('LayersConfig loaded from backend is not an defined or is empty');
+                }
+            }).catch((error) => {
+                const message = 'Error while loading layers config from backend';
+                console.error(message, error);
+                reject(message);
+        })
+    })
+}
+
```
[the remark this answers is not in the document]
Comment (pakb) [reply src/api/layers.api.js:170]: Ok then the "separate file for API call" is even more needed so that it can be changed in the future more easily.

File: src/api/layers.api.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        }
+        switch(type.toLowerCase()) {
+            case 'wmts':
+                layer = new WMTSLayer(name, id, opacity, format, timeEnableConfig, background);
+                break;
+            case 'wms':
+                layer = new WMSLayer(name, id, opacity, layerConfig.wmsUrl, format);
+                break;
+            default:
+                layer = new GeoJsonLayer(name, id, opacity);
```
[the remark this answers is not in the document]
Comment (pakb) [reply src/api/layers.api.js:138]: Yep, forgot to fix that, good catch

File: src/api/layers.api.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+import axios from "axios";
+
+/**
+ * @readonly
+ * @enum {String}
+ */
+export const LayerTypes = {
+    WMTS: 'wmts',
+    WMS: 'wms',
+    GEOJSON: 'geojson'
```
[the remark this answers is not in the document]
Comment (pakb) [reply src/api/layers.api.js:11]: Yes, I think I'll then implement a new `class` for that type of layers too

File: cypress/integration/legacyParamImport_spec.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        readStoreValue('getters.centerEpsg4326').should(center => {
+            expect(center[0]).to.eq(lon);
+            expect(center[1]).to.eq(lat);
+        });
+    });
+
+    it('reproject legacy EPSG:2056 coordinates param to EPSG:4326', () => {
+        const E = 2660000;
+        const N = 1200000;
+        const legacyZoom = 8;
```
[the remark this answers is not in the document]
Comment (pakb) [reply cypress/integration/legacyParamImport_spec.js:46]: as state below, it's a better idea than calling it `legacy`, I'll change that

File: src/modules/menu/components/MenuLangSelector.vue (hunk 1 above)
Comment (boecklic) [reply src/modules/menu/components/MenuLangSelector.vue:11]: :+1: if possible

File: src/api/layers.api.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (boecklic) [reply src/api/layers.api.js:170]: jep!

File: src/api/layers.api.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (boecklic) [reply src/api/layers.api.js:11]: :+1:

File: src/modules/menu/components/MenuLangSelector.vue (hunk 1 above)
Comment (pakb) [reply src/modules/menu/components/MenuLangSelector.vue]: So in the end I've separated E2E testing (which doesn't require to get a portion of the app) and Unit testing into two categories of test. This way I could keep the `@` notation (and it made a bit more sense...)

**Review by boecklic — APPROVED** [review]
